packages/gateway/src/server/router.ts

```diff
@@ -105,15 +105,24 @@ async function buildApi(toRouter: Router, apiDef: ApiDef, mesh?: IMesh): Promise
   );
 }
 
-async function getMeshForApiDef(apiDef: ApiDef, mesh?: IMesh, retry = 5): Promise<IMesh | undefined> {
+export async function getMeshForApiDef(
+  apiDef: ApiDef,
+  mesh?: IMesh,
+  retry = 5,
+  errorHandler?: (error: Error) => any
+): Promise<IMesh | undefined> {
   if (mesh) {
     return mesh;
   }
   try {
     const meshConfig = await processConfig({ sources: apiDef.sources, ...apiDef.mesh });
     mesh = await getMesh(meshConfig);
   } catch (error) {
-    logger.error(error);
+    if (errorHandler) {
+      errorHandler(error);
+    } else {
+      logger.error(error);
+    }
     if (retry) {
       logger.warn('Failed to load schema, retrying after 5 seconds...');
```

Comment:
will it work if included in the dashboard bk?

Reply:
no, as I pass `retry = 0` argument